### PR TITLE
JSON page that shows submissions

### PR DESCRIPTION
**Pull request by jruo**

### CODE REVIEW

File: app/controllers/courses_controller.rb [hunk 1]
```diff
@@ -70,6 +70,22 @@ def show_json
     render json: data.to_json
   end
 
+  # This is for analysis and better points from exercise program, which will be done one day.
+  def submissions_data
```
Comment (jamox) [inline app/controllers/courses_controller.rb]: Name can be too easily confused with [submission_data.rb](https://github.com/testmycode/tmc-server/blob/master/app/models/submission_data.rb)

File: lib/exercise_list.rb [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+        id: exercise.id,
+        exercise_name: exercise.name,
+        course_id: exercise.course_id,
+        available_points: AvailablePointList.new.available_point_list(exercise.available_points),
+        created_at: exercise.created_at,
+        updated_at: exercise.updated_at,
+        publish_time: exercise.publish_time,
+        hidden: exercise.hidden,
+        solution_visible_after: exercise.solution_visible_after,
+        has_tests: exercise.has_tests,
```
Comment (jamox) [inline lib/exercise_list.rb]: No need for this field yet

File: lib/exercise_list.rb [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    {
+        id: exercise.id,
+        exercise_name: exercise.name,
+        course_id: exercise.course_id,
+        available_points: AvailablePointList.new.available_point_list(exercise.available_points),
+        created_at: exercise.created_at,
+        updated_at: exercise.updated_at,
+        publish_time: exercise.publish_time,
+        hidden: exercise.hidden,
+        solution_visible_after: exercise.solution_visible_after,
```
Comment (jamox) [inline lib/exercise_list.rb]: why?

File: lib/exercise_list.rb [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+        has_tests: exercise.has_tests,
+        deadline_spec: exercise.deadline_spec,
+        soft_deadline_spec: exercise.soft_deadline_spec,
+        unlock_spec: exercise.unlock_spec,
+        code_review_requests_enabled: exercise.code_review_requests_enabled,
+        run_tests_locally_action_enabled: exercise.run_tests_locally_action_enabled
```
Comment (jamox) [inline lib/exercise_list.rb]: I don't see need for all of these fields? -- I'd avoid exposing data which is meant to be used internally without a clear need/external consumer for that information.

File: app/controllers/courses_controller.rb [hunk 5]
```diff
@@ -70,6 +70,22 @@ def show_json
...
+    authorize! :teach, @organization
+
+    submissions = @course.submissions.order('created_at DESC')
+    exercises = @course.exercises
+    data = {
+        api_version:                      ApiVersion::API_VERSION,
+        course_name:                      @course.name,
+        course_id:                        @course.id,
+        exercises:                        ExerciseList.new.exercise_list(exercises),
+        submissions:                      SubmissionList.new(current_user, view_context).submission_list_for_analysis(submissions)
```
Comment (mpartel) [inline app/controllers/courses_controller.rb]: Too much indentation.

File: lib/available_point_list.rb [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+  def available_point_list(available_points)
+    available_points.map { |ap| available_point_data(ap) }
+  end
+
+  def available_point_data(available_point)
+    {
+        id: available_point.id,
+        name: available_point.name,
+        exercise_id: available_point.exercise_id,
+        requires_review: available_point.requires_review
```
Comment (mpartel) [inline lib/available_point_list.rb]: Too much indentation.

**Comment by jamox** [comment]
I don't see clear use for this; if for a system which uses points/amount of submissions, what we should be doing is more fetching points for given user, or somehow crawl snapshot once and then use a hook in when submission is graded, to notify the app of the received submission.

And this would be a huge issue for courses where we have say 200 000 submissions.

And we already have api `courses.json` and `exercises/:id.json` which povides more or less all the really needed fields...

**Pull request closed without merging**